src/common.c

```diff
@@ -367,6 +376,25 @@ static int ofi_str_to_psmx2(const char *str, void **addr, size_t *len)
 	return -FI_EINVAL;
 }
 
+static int ofi_str_to_verbs_ud(const char *str, void **addr, size_t *len)
+{
+	int ret;
+
+	*len = 4 * sizeof(uint64_t);
+	if (!(*addr = calloc(1, *len)))
+		return -FI_ENOMEM;
+
+	ret = sscanf(str, "%*[^:]://%" SCNx64 ":%" SCNx64
+			  ":%" SCNx64 ":%" SCNx64,
+		     (uint64_t *)*addr, (uint64_t *)*addr + 1,
+		     (uint64_t *)*addr + 2, (uint64_t *)*addr + 3);
```

Comment:
Check out https://en.wikipedia.org/wiki/Uniform_Resource_Identifier.

If the `:` is in the authority, it appears to be ok.  If the `:` is in the path part, then it needs to be `%`-encoded (i.e., `%25`).